include/lcp-paginator.php

```diff
@@ -149,4 +150,25 @@ private function lcp_page_link($page, $current_page, $lcp_instance, $bookmark, $
     $link = str_replace('?&', '?', $link );
     return $link;
   }
+
+  private function process_query_params($query) {
+    parse_str($query, $params);
+
+    // Filter out all none page parameters.
+    $params = array_filter($params, function ($k) {
+      if (strpos($k, 'page', 0)) {
+          return true;
+      }
+      return false;
+    }, ARRAY_FILTER_USE_KEY);
```

Comment:
You can't just discard all non page related query parameters! You don't know what particular setup a website might be using, there may be (many) other important query parameters that we shouldn't touch, one simple example is when a WP page is not using pretty permalinks - your code would not work in such a case. Please adjust your code so that it validates the `lcp_page<instance>` and doesn't touch any other query parameters.

Reply:
Noted, will work on resolving these issues when I have some free time later this week.